### ISSUE TITLE
ILVerify complains about conversion from `string[]` to `IList<object>`

### ISSUE THREAD

**Issue by jcouv** [dotnet/runtime#106282]
I found this issue while running the test below from roslyn. It shows the original C# code, the generated IL and the ILVerify error (```Unexpected type on the stack. { Offset = 0x1, Found = ref 'string[]', Expected = ref '[netstandard]System.Collections.Generic.IList`1<object>' }```).

I would expect no ILVerify diagnostic for this IL.

ECMA-335 states:
> I.8.7.1 Assignment compatibility for signature types
> A signature type T is compatible-with a signature type U if and only if at least one of the 
following holds:
> ...
> 3. T is a reference type, and U is the direct base class of T. 
> ...
> 7. T is a zero-based rank-1 array V[], and U is IList<W>, and V is array-elementcompatible-with W.

> A signature type T is array-element-compatible-with a signature type U if and only if T has 
underlying type V and U has underlying type W and either: 
> 1. V is compatible-with W; or 
> 2. V and W have the same reduced type.

```csharp
    [Fact]
    public void TODO2()
    {
        var src3 = """
System.Console.Write(C.Method(new string[] { "ran" })[0]);

public class C
{
    public static System.Collections.Generic.IList<object> Method(string[] s) => s;
}
""";
        var comp3 = CreateCompilation(src3);
        // [Method]: Unexpected type on the stack. { Offset = 0x1, Found = ref 'string[]', Expected = ref '[netstandard]System.Collections.Generic.IList`1<object>' }
        var verifier3 = CompileAndVerify(comp3, expectedOutput: "ran", verify: Verification.FailsILVerify).VerifyDiagnostics();
        verifier3.VerifyIL("C.Method", """
{
  // Code size        2 (0x2)
  .maxstack  1
  IL_0000:  ldarg.0
  IL_0001:  ret
}
""");
    }
```

**Comment by MichalStrehovsky** [comment]
@jcouv What version of ILVerify is this?

**Comment by jcouv** [comment]
Sorry, I should have provided that information. This is on ILVerify 8.0.0-rtm.23523.3

**Comment by MichalStrehovsky** [comment]
It's likely fixed in 9.0. Can you try with a 9.0 build?

**Comment by jcouv** [comment]
I gave it a shot for 20 minutes but that change snowballing. Let me close this issue and track this on the roslyn side. 
When we update to 9.0, I will re-open this if the problem still occurs. Thanks
